src/main/java/com/google/sps/servlets/RecipeServlet.java

```diff
@@ -15,27 +15,55 @@
 package com.google.sps.servlets;
 
 import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.lang.StringBuilder;
+import java.io.BufferedReader;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
+import javax.servlet.AsyncContext;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.cloud.firestore.CollectionReference;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.sps.data.Recipe;
+import com.google.sps.data.DBReferences;
 
 /** Handles recipe post requests. */
 @WebServlet("/api/post")
-public class RecipeServlet extends HttpServlet {
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {}
+public class RecipeServlet extends HttpServlet
+{
 
-  @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {}
+    private Gson gson = new Gson();
 
-  @Override
-  public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {}
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {}
 
-  @Override
-  public void doDelete(HttpServletRequest request, HttpServletResponse response)
-      throws IOException {}
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException
+    {
+        String data = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
+        Recipe newRecipe = gson.fromJson(data, Recipe.class);
+        if(newRecipe.content == null || newRecipe.title == null)
+        {
```

Comment:
Style nit: Move { to end of previous line.

Reply:
Done in e26d3b9